packages/components/pie-button/test/helpers/page-object/pie-button-form-integration.page.ts

```diff
@@ -0,0 +1,188 @@
+        await this.userPasswordInput.fill(inputData.userPassword);
+        await this.userPaymentCardTypeSelect.selectOption(inputData.userPaymentCardType);
+        await this.userPaymentCardNumberInput.fill(inputData.userPaymentCardNumber);
+        await this.userPaymentCardExpirationInput.fill(inputData.userPaymentCardExpiration);
+
+        return this;
+    }
+
+    /**
+     * Clicks the specified form button.
+     *
+     * @param {FormButtonType} buttonType - The type of the button to click.
+     * @returns {Promise<void>} A Promise that resolves once the button has been clicked.
+     */
+    async clickFormButton (buttonType: FormButtonType): Promise<void> {
+        switch (buttonType) {
+            case 'pie-button-submit':
+                await this.submitPieButton.click();
+                break;
+            case 'pie-button-reset':
+                await this.resetPieButton.click();
+                break;
+            case 'native-reset':
+                await this.resetNativeButton.click();
+                break;
+            default:
+                throw new Error(`Invalid button type: ${buttonType}`);
+        }
+    }
+
+    /**
+     * Focuses on the specified field.
+     *
+     * @param {keyof FormInput} fieldName - The name of the field to focus on.
+     * @returns {Promise<Page>} A Promise that resolves once the field has been focused.
+     */
+    async focusField (fieldName: keyof FormInput): Promise<Locator> {
+        switch (fieldName) {
+            case 'userName':
+                await this.userNameInput.focus();
+                return this.userNameInput;
+            case 'userEmail':
+                await this.userEmailInput.focus();
+                return this.userEmailInput;
+            case 'userPassword':
+                await this.userPasswordInput.focus();
+                return this.userPasswordInput;
+            case 'userPaymentCardType':
+                await this.userPaymentCardTypeSelect.focus();
+                return this.userPaymentCardTypeSelect;
+            case 'userPaymentCardNumber':
+                await this.userPaymentCardNumberInput.focus();
+                return this.userPaymentCardNumberInput;
+            case 'userPaymentCardExpiration':
+                await this.userPaymentCardExpirationInput.focus();
+                return this.userPaymentCardExpirationInput;
+            default:
+                throw new Error(`Invalid field name: ${fieldName}`);
+        }
+    }
+
+    /**
+     * Focuses on the specified button.
+     *
+     * @param {FormButtonType} formButtonType - The type of the button to focus on.
+     * @returns {Promise<Locator>} A Promise that resolves once the button has been focused.
+     */
+    async focusButton (buttonType: FormButtonType): Promise<Locator> {
+        switch (buttonType) {
+            case 'pie-button-submit':
+                await this.submitPieButton.focus();
+                return this.submitPieButton;
```

Comment:
I don't quite understand why we return the button, but could we not simply return `this.submitPieButton.focus()`? Because this entire method is returning a promise, I don't really know why it needs to `await` anything inside. The caller can await this result